plugins/woocommerce/includes/class-wc-comments.php

```diff
@@ -259,8 +259,13 @@ public static function comment_moderation_recipients( $emails, $comment_id ) {
 	 * @param int $post_id Post ID.
 	 */
 	public static function clear_transients( $post_id ) {
-		if ( 'product' === get_post_type( $post_id ) ) {
-			$product = wc_get_product( $post_id );
+		$post_id = absint( $post_id );
+		if ( $post_id === 0 || 'product' !== get_post_type( $post_id ) ) {
+			return;
+		}
```

Comment:
hmm, although `get_post_type` will attempt to get a post from the `$post` global [if it is falsey](https://github.com/WordPress/wordpress-develop/blob/700708188a4517aac27f2e522050aaf2112e2240/src/wp-includes/post.php#L1096-L1098). So I wonder if the initial check here should be a similar falsey check?

Reply:
nvm the `absint` is here 🤦🏻